schemas/schema-schema.ipldsch

```diff
@@ -148,30 +148,131 @@ type AnyScalar union {
 	| Float float
 } representation kinded
 
+## AdvancedDataLayout holds `advanced` definitions which currently only require
+## a name.
+##
+## Use `advanced Foo` rather than `type Foo` to indicate an advanced data layout
+## (ADL) with that name which can be used as a representation for type
+## definitions whose kind the ADL is able to support.
+##
+## The ADL name is currently the only identifier that can be used to make a
+## connection with the algorithm/logic behind this ADL. Future iterations may
+## formalize this connection by some other means.
+##
+type AdvancedDataLayout struct {
+	name String
+}
+
+## AdvancedRepresentation allows for "advanced" representations on all of
+## basic types. Currently it only requires a "name", which is supplied short-
+## hand after the `advanced` keyword
+## (e.g. `type Foo string representation advanced Bar`) where "Bar" is the
+## name. This name should match an ADL definition that exists elsewhere in the
+## schema.
+##
+type AdvancedRepresentation struct {
+	name String
```

Comment:
Would recommend adding a `type AdvancedLayoutRef string` type and using that here (and then as well in other places it occurs).

Should show up roughly two places (which should make sense because it's sort of like a primary key in one of the positions, and a foreign key reference in the other, roughly speaking).